store/task.go

```diff
@@ -640,56 +640,52 @@ func (s *Store) patchTaskStatusImpl(ctx context.Context, tx *sql.Tx, patch *api.
 		return nil, &common.Error{Code: common.NotFound, Err: fmt.Errorf("task ID not found: %d", patch.ID)}
 	}
 
-	if !(taskRawObj.Status == api.TaskPendingApproval && patch.Status == api.TaskPending) {
-		taskRunFind := &api.TaskRunFind{
-			TaskID: &taskRawObj.ID,
-			StatusList: &[]api.TaskRunStatus{
-				api.TaskRunRunning,
-			},
+	taskRunFind := &api.TaskRunFind{
+		TaskID: &taskRawObj.ID,
+		StatusList: &[]api.TaskRunStatus{
+			api.TaskRunRunning,
+		},
+	}
+	taskRunRaw, err := s.getTaskRunRawTx(ctx, tx, taskRunFind)
+	if err != nil {
+		return nil, err
+	}
+	if taskRunRaw == nil && patch.Status == api.TaskRunning {
```

Comment:
This is actually an UPSERT semantic instead of PATCH.
Maybe we should come to this later and see if it's appropriate to refactor.